src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -323,33 +278,32 @@ def _delegation(self, jobID):
             params = {"action": "new"}
             query = self.base_url + command
             proxy = X509Chain()
-            res = proxy.loadProxyFromFile(self.session.cert)
-            r = self.session.post(
+            res = proxy.loadProxyFromFile(self.s.cert)
+            r = self.s.post(
                 query, data=proxy.dumpAllToString(), headers=self.headers, params=params, timeout=self.arcRESTTimeout
             )
             dID = ""
             if r.ok:  # Get the delegation and "PUT" it in the CE ...
                 dID = r.headers.get("location", "")
-                data = proxy.generateChainFromRequestString(r.text, lifetime=4 * 12 * 3600)["Value"]
                 if len(dID) > 2:
                     dID = dID.split("new/")[-1]
                     command = "delegations/" + dID
                     query = self.base_url + command
-                    r1 = self.session.put(query, data=data, headers=self.headers, timeout=self.arcRESTTimeout)
+                    r1 = self.s.put(query, data=r.text, headers=self.headers, timeout=self.arcRESTTimeout)
                     if not r1.ok:
                         dID = ""
                 else:
                     dID = ""
             return dID
         else:  # Retrieve delegation for existing job
-            jobsJson = {"job": [{"id": jobID}]}  # job in ARC REST json format
+            jj = {"job": [{"id": jobID}]}  # job in ARC REST json format
             command = "jobs"
             params = {"action": "delegations"}
             query = self.base_url + command
-            r = self.session.post(query, data=json.dumps(jobsJson), headers=self.headers, timeout=self.arcRESTTimeout)
+            r = self.s.post(query, data=json.dumps(jj), headers=self.headers, timeout=self.arcRESTTimeout)
             dID = ""
             if r.ok:  # Check if the job has a delegation
-                p = r.json()
+                p = json.loads(r.text.replace("\n", ""))
```

Comment:
Why is this better than `r.json()`?

Reply:
Old history - until we got the ARC developers to fix issues with their response strings, the replace was essential to prevent crashes (see earlier thread of my discussion with Chris Burr and bug report on nordugrid bugzilla).

Reply:
Ok, could you add a comment about that above this line then, please?

Reply:
This bug was fixed so it should be using `r.json()`.

Reply:
Yes - agreed.

Reply:
Done.